lib/src/fixed_datetime_formatter.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+/// A class for parsing and formatting dates for a fixed format string. For
```

Comment:
[nit] The "A class" is redundant. Also, add a blank line following the header sentence.

https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

How about

/// A formatter and parser for Dates in a fixed format string.
///
/// For example, calling